typescript/utils/src/validator.ts

```diff
@@ -1,47 +1,64 @@
-import { BigNumberish, ethers } from 'ethers';
+import { ethers } from 'ethers';
 
 import { types, utils } from '@abacus-network/utils';
 
-export class Validator {
+import { Checkpoint } from './types';
+
+export class BaseValidator {
   localDomain: types.Domain;
-  signer: ethers.Signer;
   address: types.Address;
 
-  constructor(
-    signer: ethers.Signer,
-    address: types.Address,
-    localDomain: types.Domain,
-    disableWarn: boolean,
-  ) {
-    if (!disableWarn) {
-      throw new Error('Please use `Validator.fromSigner()` to instantiate.');
-    }
-    this.localDomain = localDomain ? localDomain : 0;
-    this.signer = signer;
+  constructor(address: types.Address, localDomain: types.Domain) {
+    this.localDomain = localDomain;
     this.address = address;
   }
 
-  static async fromSigner(signer: ethers.Signer, localDomain: types.Domain) {
-    return new Validator(signer, await signer.getAddress(), localDomain, true);
-  }
-
   domainHash() {
     return utils.domainHash(this.localDomain);
   }
 
-  message(root: types.HexString, index: BigNumberish) {
+  message(root: types.HexString, index: number) {
     return ethers.utils.solidityPack(
       ['bytes32', 'bytes32', 'uint256'],
       [this.domainHash(), root, index],
     );
   }
 
-  async signCheckpoint(root: types.HexString, index: BigNumberish) {
+  messageHash(root: types.HexString, index: number) {
     const message = this.message(root, index);
-    const msgHash = ethers.utils.arrayify(ethers.utils.keccak256(message));
+    return ethers.utils.arrayify(ethers.utils.keccak256(message));
+  }
+
+  recoverAddressFromCheckpoint(checkpoint: Checkpoint): types.Address {
+    const msgHash = this.messageHash(checkpoint.root, checkpoint.index);
+    return ethers.utils.recoverAddress(msgHash, checkpoint.signature);
+  }
+
+  matchesSigner(checkpoint: Checkpoint) {
+    return this.recoverAddressFromCheckpoint(checkpoint) === this.address;
+  }
+}
+
+export class Validator extends BaseValidator {
```

Comment:
Why split this into another validator class? I am having trouble understanding how this class is conceptually different.

Reply:
I didn't want to change all of the existing imports
this is non-breaking